### PR TITLE
WT-9171 Add a dist/ script that detects files with no newline at the end.

### PR DESCRIPTION
**Pull request by ruby-oujo**
* Adding newlines at the end of files in the absence of one in `s_whitespace`. 
* When a newline is added and during the running of `./s_all` an informative statement is printed.

### CODE REVIEW

File: dist/s_whitespace [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
 	! head $1 | grep -q 'automatically generated by SWIG' || return
 	sed -e 's/[	 ][	 ]*$//' < $1 | \
 	    cat -s | \
 	    sed -e '${' -e '/^$/d' -e '}' > $t
 	cmp $t $1 > /dev/null 2>&1 || (echo "$1" && cp $t $1)
+    if (test "$(tail -c 1 "$1" | wc -l)" -eq 0)
+    then
+        echo "Adding newline at end of file: $f"
+    fi
+    sed -i -e '$a\' $1
```
Comment (luke-pearson) [inline dist/s_whitespace:21]: So we just run this for everything and presumably it only modifies the things it needs to? Does its return code indicate whether it did anything?

Comment (ruby-oujo) [reply dist/s_whitespace:21]: Yes this runs on everything but it only adds a newline to the end of a file if there isn't one. So if a file has a newline nothing will be added. There is no return code but I used the above check to see if a file does not have a newline; so any file that has been modified will be printed.

Comment (ruby-oujo) [reply dist/s_whitespace]: Thanks @keithbostic ! I have updated the function with those changes.

**Review by luke-pearson — APPROVED** [review]
lgtm.

**Review by jeremythorp — COMMENTED** [review]
LGTM too!

**Pull request merged**